youtube_dl/extractor/iheartradio.py

```diff
@@ -0,0 +1,213 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import json
+import re
+import uuid
+
+from .common import InfoExtractor
+
+from ..compat import compat_str
+
+from numbers import Number
+
+from ..utils import (
+    clean_html,
+    urlencode_postdata
+)
+
+
+# To get the audio files, we have to use their internal API
+class IHeartRadioPodcastBaseIE(InfoExtractor):
+    # Register anonymous user, same behavior as web app
+    def _register_temp_user(self, current_id):
+        random_device_id = compat_str(uuid.uuid4())
+        random_oauth_id = compat_str(uuid.uuid4())
+
+        register_user_values = urlencode_postdata({
+            'accessToken': 'anon',
+            'accessTokenType': 'anon',
+            'deviceId': random_device_id,
+            'deviceName': "web-desktop",
+            'host': "webapp.WW",
+            'oauthUuid': random_oauth_id,
+            'userName': 'anon' + random_oauth_id
+        })
+        return self._download_json(
+            'https://ww.api.iheart.com/api/v1/account/loginOrCreateOauthUser',
+            current_id, "Registering temporary user", data=register_user_values,
+            headers={'Accept': 'application/json, text/plain, */*',
+                     'X-hostName': 'webapp.WW'})
```

Comment:
I'm not sure how I would do that, I was just looking at the network requests done by the web frontend and copying that behavior.

Also, what are we trying to avoid? Other API calls require the output from this.

Reply:
> I'm not sure how I would do that, I was just looking at the network requests done by the web frontend and copying that behavior.
> 
> Also, what are we trying to avoid? Other API calls require the output from this.

trying to avoid additional requests, I think to move quickly with this PR, I would make a simpler implementation, and then you can base your modification on it.

Reply:
the extractor that relies on API v3 has been added in 9c484c0019ba16cd4b4d686930bfa038f1bf9320.